acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/acceptance/jsonrpc/ExecutionEngineAcceptanceTest.java

```diff
@@ -14,96 +14,38 @@
-import java.util.Arrays;
-import java.util.stream.Collectors;
 
-import com.fasterxml.jackson.databind.ObjectMapper;
-import okhttp3.Call;
-import okhttp3.MediaType;
-import okhttp3.OkHttpClient;
-import okhttp3.Request;
-import okhttp3.RequestBody;
-import okhttp3.Response;
 import org.junit.AfterClass;
 import org.junit.BeforeClass;
-import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.junit.runners.Parameterized;
 
 @RunWith(Parameterized.class)
-public class ExecutionEngineAcceptanceTest {
+public class ExecutionEngineAcceptanceTest extends AbstractJsonRpcTest {
   private static final String GENESIS_FILE = "/jsonrpc/engine/genesis.json";
   private static final String TEST_CASE_PATH = "/jsonrpc/engine/test-cases/";
-  private static final MediaType MEDIA_TYPE_JSON =
-      MediaType.parse("application/json; charset=utf-8");
-
-  private static Cluster cluster;
-  private static BesuNode executionEngine;
-  private static OkHttpClient consensusClient;
-  private static ObjectMapper mapper;
 
-  private final URI testCaseFileURI;
+  private static JsonRpcTestsContext testsContext;
 
   public ExecutionEngineAcceptanceTest(final String ignored, final URI testCaseFileURI) {
-    this.testCaseFileURI = testCaseFileURI;
+    super(ignored, testsContext, testCaseFileURI);
   }
 
   @BeforeClass
   public static void init() throws IOException {
-    cluster = new Cluster(new NetConditions(new NetTransactions()));
-
-    executionEngine =
-        new BesuNodeFactory().createExecutionEngineGenesisNode("executionEngine", GENESIS_FILE);
-    cluster.start(executionEngine);
-    consensusClient = new OkHttpClient();
-
-    mapper = new ObjectMapper();
-  }
-
-  @Test
-  public void test() throws IOException {
-    final EngineTestCase testCase = mapper.readValue(testCaseFileURI.toURL(), EngineTestCase.class);
-
-    final Call preparePayloadRequest =
-        consensusClient.newCall(
-            new Request.Builder()
-                .url(executionEngine.engineRpcUrl().get())
-                .post(RequestBody.create(testCase.getRequest().toString(), MEDIA_TYPE_JSON))
-                .build());
-    final Response response = preparePayloadRequest.execute();
-
-    assertThat(response.code()).isEqualTo(testCase.getStatusCode());
-    assertThat(response.body().string()).isEqualTo(testCase.getResponse().toPrettyString());
+    testsContext = new JsonRpcTestsContext(GENESIS_FILE);
   }
 
   @Parameterized.Parameters(name = "{0}")
   public static Iterable<Object[]> testCases() throws URISyntaxException {
-
-    final File testCasePath =
-        new File(ExecutionEngineAcceptanceTest.class.getResource(TEST_CASE_PATH).toURI());
-    final File[] testCasesList = testCasePath.listFiles();
-
-    return Arrays.stream(testCasesList)
-        .sorted()
-        .map(file -> new Object[] {file.getName(), file.toURI()})
-        .collect(Collectors.toList());
+    return AbstractJsonRpcTest.testCases(TEST_CASE_PATH);
```

Comment:
Not a big fan of this static method approach, appreciate it's not much different from what's there already. Wonder if there's a better way using composition?